dashboard/db/migrate/20160818190145_add_unique_constraint_to_pd_workshop_attendance.rb

```diff
@@ -0,0 +1,27 @@
+class AddUniqueConstraintToPdWorkshopAttendance < ActiveRecord::Migration
+  def change
+    reversible do |dir|
+      dir.up do
+        duplicate_attendance_values = Pd::Attendance.
+          select('pd_session_id, teacher_id, count(*)').
+          group(:pd_session_id, :teacher_id).
+          having('count(*) > 1').
+          pluck(:pd_session_id, :teacher_id)
+
+        # Destroy all but the latest of each set
+        duplicate_attendance_values.each do |pd_session_id, teacher_id|
+          Pd::Attendance.
+            where(pd_session_id: pd_session_id, teacher_id: teacher_id).
+            order(id: :desc)[1..-1].
+            each(&:destroy)
+        end
+      end
+    end
+
+    # Add unique multi-column index and remove old single-column index
+    change_table :pd_attendances do |t|
+      t.index [:pd_session_id, :teacher_id], unique: true
```

Comment:
Pretty sure this should be `t.add_index` (http://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/change_table). Please run this migration up and down, confirming that it removes all but the most recent in case of duplicates.